### PR TITLE
Docs: Add limitations for Basic Authentication

### PR DESCRIPTION
**Pull request by hmanwani-rh**
See JIRA: https://issues.redhat.com/browse/QDOCS-29

Added limitations for Basic Authentication in "security-built-in-authentication.adoc" file. 

Doc preview: http://file.pnq.redhat.com/~hmanwani/QDOCS-29-main

### CODE REVIEW

File: docs/src/main/asciidoc/security-built-in-authentication.adoc [hunk 1]
```diff
@@ -23,7 +23,29 @@ If the user name is `Alice` and the password is `secret`, the HTTP authorization
 ====
  
 The Basic Authentication mechanism does not provide confidentiality protection for the transmitted credentials. The credentials are merely encoded with Base64 when in transit and not encrypted or hashed in any way. Therefore, Basic Authentication is used with HTTPS to provide confidentiality.
- 
+
+There are a few limitations associated with Basic Authentication, which include:
+
+Credentials as plain text::
```
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-built-in-authentication.adoc:29]:
```suggestion
Credentials are sent as plain text::
```

File: docs/src/main/asciidoc/security-built-in-authentication.adoc [hunk 2]
```diff
@@ -23,7 +23,29 @@ If the user name is `Alice` and the password is `secret`, the HTTP authorization
...
++
+--
+It is required to use HTTPS with Basic Authentication. However, if the load balancer terminates HTTPS, the risk of exposing credentials as plain text increases when HTTP is forwarded to Quarkus.
+
+Also, in multi-hop deployments, HTTPS is only used between client and the first Quarkus endpoint, and the basic credentials are propagated from the first Quarkus endpoint to the next endpoint. Therefore, the credentials are sent as plain text.
```
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-built-in-authentication.adoc:34]:
```suggestion
Also, in multi-hop deployments, the credentials can be exposed if HTTPS is used between the client and the first Quarkus endpoint only but they are propagated to the next endpoint over HTTP.
```

File: docs/src/main/asciidoc/security-built-in-authentication.adoc [hunk 3]
```diff
@@ -23,7 +23,29 @@ If the user name is `Alice` and the password is `secret`, the HTTP authorization
...
+Credentials being exposed::
++
+--
+In Basic Authentication, a username and password need to be sent with each request, which increases the risk of credentials being exposed.
+--
+
+Application complexity increases::
++
+--
+The Quarkus application needs to ensure that username, passwords, and roles are securely secured, which increases the application complexity.
```
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-built-in-authentication.adoc:46]:
```suggestion
The Quarkus application needs to ensure itself that the usernames, passwords, and roles are managed securely, which increases the application complexity. 
```

**Comment by sberyozkin** [comment]
Hi @hmanwani-rh Nice work, I've added a few minor suggestions which I thought could clarify a few details better, have a look please

File: docs/src/main/asciidoc/security-built-in-authentication.adoc [hunk 4]
```diff
@@ -23,7 +23,29 @@ If the user name is `Alice` and the password is `secret`, the HTTP authorization
...
  
 The Basic Authentication mechanism does not provide confidentiality protection for the transmitted credentials. The credentials are merely encoded with Base64 when in transit and not encrypted or hashed in any way. Therefore, Basic Authentication is used with HTTPS to provide confidentiality.
- 
+
+There are a few limitations associated with Basic Authentication, which include:
+
+Credentials as plain text::
++
+--
+It is required to use HTTPS with Basic Authentication. However, if the load balancer terminates HTTPS, the risk of exposing credentials as plain text increases when HTTP is forwarded to Quarkus.
```
[the remark this answers is not in the document]
Comment (hmanwani-rh) [reply docs/src/main/asciidoc/security-built-in-authentication.adoc:32]: Done, added the suggestion.

File: docs/src/main/asciidoc/security-built-in-authentication.adoc (hunk 2 above)
Comment (hmanwani-rh) [reply docs/src/main/asciidoc/security-built-in-authentication.adoc:34]: Done, changed the sentence as follows:

"Also, in multi-hop deployments, the credentials can be exposed if HTTPS is used between the client and the first Quarkus endpoint only. However, the credentials are propagated to the next Quarkus endpoint over HTTP."

File: docs/src/main/asciidoc/security-built-in-authentication.adoc (hunk 3 above)
Comment (hmanwani-rh) [reply docs/src/main/asciidoc/security-built-in-authentication.adoc:46]: Done, added the suggestion.

File: docs/src/main/asciidoc/security-built-in-authentication.adoc [hunk 5]
```diff
@@ -23,7 +23,29 @@ If the user name is `Alice` and the password is `secret`, the HTTP authorization
...
+
+Also, in multi-hop deployments, HTTPS is only used between client and the first Quarkus endpoint, and the basic credentials are propagated from the first Quarkus endpoint to the next endpoint. Therefore, the credentials are sent as plain text.
+--
+
+Credentials being exposed::
```
[the remark this answers is not in the document]
Comment (hmanwani-rh) [reply docs/src/main/asciidoc/security-built-in-authentication.adoc:37]: Done, added the suggestion

File: docs/src/main/asciidoc/security-built-in-authentication.adoc (hunk 1 above)
Comment (hmanwani-rh) [reply docs/src/main/asciidoc/security-built-in-authentication.adoc:29]: Done, added the suggestion

**Comment by hmanwani-rh** [comment]
> Hi @hmanwani-rh Nice work, I've added a few minor suggestions which I thought could clarify a few details better, have a look please

@sberyozkin Thanks for reviewing the update and sharing the comments. I've incorporated the changes as suggested.

**Comment by MichalMaler** [comment]
> > Hi @hmanwani-rh Nice work, I've added a few minor suggestions which I thought could clarify a few details better, have a look please
> 
> @sberyozkin Thanks for reviewing the update and sharing the comments. I've incorporated the changes as suggested.

Good Job @hmanwani-rh . Will add my review now. Will add just few lines to make it complete.

File: docs/src/main/asciidoc/security-built-in-authentication.adoc [hunk 6]
```diff
@@ -23,7 +23,29 @@ If the user name is `Alice` and the password is `secret`, the HTTP authorization
 ====
  
 The Basic Authentication mechanism does not provide confidentiality protection for the transmitted credentials. The credentials are merely encoded with Base64 when in transit and not encrypted or hashed in any way. Therefore, Basic Authentication is used with HTTPS to provide confidentiality.
- 
+
+There are a few limitations associated with Basic Authentication, which include:
```
Comment (MichalMaler) [inline docs/src/main/asciidoc/security-built-in-authentication.adoc:27]:
```suggestion
Basic Authentication is a well-specified challenge/response scheme that all browsers and most Web servers understand. The main benefit of this method is its ability to work through proxy servers. However, there are a few limitations associated with Basic Authentication, which include:
```

File: docs/src/main/asciidoc/security-built-in-authentication.adoc [hunk 7]
```diff
@@ -23,7 +23,29 @@ If the user name is `Alice` and the password is `secret`, the HTTP authorization
...
+Credentials are sent with each request::
++
+--
+In Basic Authentication, a username and password need to be sent with each request, which increases the risk of credentials being exposed.
+--
+
+Application complexity increases::
++
+--
+The Quarkus application needs to ensure itself that usernames, passwords, and roles are managed securely, which increases the application complexity.
```
Comment (MichalMaler) [inline docs/src/main/asciidoc/security-built-in-authentication.adoc:46]:
```suggestion
The Quarkus application needs to ensure that usernames, passwords, and roles are managed securely, which increases the application complexity.
```

File: docs/src/main/asciidoc/security-built-in-authentication.adoc (hunk 6 above)
Comment (hmanwani-rh) [reply docs/src/main/asciidoc/security-built-in-authentication.adoc:27]: Thanks for sharing this. 
Added this information.

File: docs/src/main/asciidoc/security-built-in-authentication.adoc (hunk 7 above)
Comment (hmanwani-rh) [reply docs/src/main/asciidoc/security-built-in-authentication.adoc:46]: I think I'll keep it as is to force on the statement that this actually increases the application complexity.

Comment (MichalMaler) [reply docs/src/main/asciidoc/security-built-in-authentication.adoc:46]: OK :)

**Review by MichalMaler — APPROVED** [review]

**Comment by MichalMaler** [comment]
LGTM

File: docs/src/main/asciidoc/security-built-in-authentication.adoc [hunk 8]
```diff
@@ -23,7 +23,30 @@ If the user name is `Alice` and the password is `secret`, the HTTP authorization
 ====
  
 The Basic Authentication mechanism does not provide confidentiality protection for the transmitted credentials. The credentials are merely encoded with Base64 when in transit and not encrypted or hashed in any way. Therefore, Basic Authentication is used with HTTPS to provide confidentiality.
- 
+
+
+Basic Authentication is a well-specified challenge or response scheme that all web browsers and most web servers understand. The main benefit of this method is its ability to work through proxy servers. However, there are a few limitations associated with Basic Authentication, which include:
```
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-built-in-authentication.adoc:28]:
```suggestion
Basic Authentication is a well-specified, simple challenge or response scheme that all web browsers and most web servers understand. However, there are a few limitations associated with Basic Authentication, which include:
```

File: docs/src/main/asciidoc/security-built-in-authentication.adoc [hunk 9]
```diff
@@ -23,7 +23,30 @@ If the user name is `Alice` and the password is `secret`, the HTTP authorization
...
+
+Credentials are sent as plain text::
++
+--
+It is required to use HTTPS with Basic Authentication to avoid exposing the credentials. However, if the load balancer terminates HTTPS, the risk of exposing credentials as plain text increases when a request is forwarded to Quarkus over HTTP.
+
+Also, in multi-hop deployments, the credentials can be exposed if HTTPS is used between the client and the first Quarkus endpoint only. However, the credentials are propagated to the next Quarkus endpoint over HTTP.
```
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-built-in-authentication.adoc:35]:
```suggestion
Also, in multi-hop deployments, the credentials can be exposed if HTTPS is used between the client and the first Quarkus endpoint only and the credentials are propagated to the next Quarkus endpoint over HTTP.
```

Comment (hmanwani-rh) [reply docs/src/main/asciidoc/security-built-in-authentication.adoc:34]: Done, added the suggestion.

File: docs/src/main/asciidoc/security-built-in-authentication.adoc [hunk 10]
```diff
@@ -23,7 +23,29 @@ If the user name is `Alice` and the password is `secret`, the HTTP authorization
 ====
  
 The Basic Authentication mechanism does not provide confidentiality protection for the transmitted credentials. The credentials are merely encoded with Base64 when in transit and not encrypted or hashed in any way. Therefore, Basic Authentication is used with HTTPS to provide confidentiality.
- 
+
+Basic Authentication is a well-specified, simple challenge or response scheme that all web browsers and most web servers understand. However, there are a few limitations associated with Basic Authentication, which include:
```
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-built-in-authentication.adoc:27]:
```suggestion
Basic Authentication is a well-specified, simple challenge and response scheme that all web browsers and most web servers understand. However, there are a few limitations associated with Basic Authentication, which include:
```

File: docs/src/main/asciidoc/security-built-in-authentication.adoc [hunk 11]
```diff
@@ -23,7 +23,29 @@ If the user name is `Alice` and the password is `secret`, the HTTP authorization
...
- 
+
+Basic Authentication is a well-specified, simple challenge and response scheme that all web browsers and most web servers understand. However, there are a few limitations associated with Basic Authentication, which include:
+
+Credentials are sent as plain text::
++
+--
+It is required to use HTTPS with Basic Authentication to avoid exposing the credentials. However, if the load balancer terminates HTTPS, the risk of exposing credentials as plain text increases when a request is forwarded to Quarkus over HTTP.
+
+Also, in multi-hop deployments, the credentials can be exposed if HTTPS is used between the client and the first Quarkus endpoint only. However, the credentials are propagated to the next Quarkus endpoint over HTTP.
```
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-built-in-authentication.adoc:34]:
```suggestion
Also, in multi-hop deployments, the credentials can be exposed if HTTPS is used between the client and the first Quarkus endpoint only and the credentials are propagated to the next Quarkus endpoint over HTTP.
```

Comment (hmanwani-rh) [reply docs/src/main/asciidoc/security-built-in-authentication.adoc]: @sberyozkin Thanks for adding that. Fixed the sentence as suggested.

**Comment by sberyozkin** [comment]
Thanks @hmanwani-rh I've just squashed a few last commits with `Minor fix` messages...

**Review by sberyozkin — APPROVED** [review]

**Comment by MichalMaler** [comment]
> @MichalMaler I kept the last 2 ones as I did not want a reference to your contribution be lost :-), I will squash now to the single commit as looks like you are OK with it :-)

Oh well, hope it will be written forever in our hearts :D :D Thank you for the attitude thought! Well done, We!

**Pull request merged**